### DISCUSSION TITLE
import notes from KeepNote

### DISCUSSION THREAD

**Discussion by tartetatin18** [TriliumNext/Trilium#2732] [Q&A] [answered]
Hi,
Does anybody knows if it's possible to import notes from KeepNote into Trilium Notes ?
If it's possible, hwo to proceed ?
Thanks in davance for your answers.

**Comment by sigaloid** [comment, marked as answer]
It seems they are stored in XML and HTML format so it should be possible to import into Trilium. Locate where the note files are stored and import by right-clicking the root note and selecting "Import".

**Reply by tartetatin18** [reply]
Thanks for your answer :)
I tried try this already, but it's just the content of the note that is imported, the title is not imported. I would have to re-name each note, one by one ...  Is there a better way ?
Could I add a tag in the HTML note (that I want to import), so that the name of note is imported at the same time ?

since the notes are stored in a db file for trilium, it gives us small possiblities of automatization ... or maybe there's something I'm not aware of ? ... and would be delighted to know about  :)

**Reply by sigaloid** [reply]
Hi, I cannot open KeepNote as it's very old. Do you mind providing a sample HTML file by creating a new note and entering a title + content?

**Reply by tartetatin18** [reply]
Thanks a LOT for your answer :) While creating the HTML file for you, the solution poped up :)
I explain : I exported this newly created note into HMTL format. In this file, there are 2 tags : 'title' and 'body', and the file was named "page.html". That's when the "bingo" occured, since yesterday I noticed that all my imported notes where named "page" ... the same name as the file name itself !
So I renamed the HTML file, and when I imported it into trilium, the note took the name of the file :)
So I just need to write a little script that will read every <title> tag/value and rename all single html file with the value of this tag and that should do it.
Thanks a lot again for your help :)
